test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -25,994 +43,1415 @@ const noop = () => {};
-              {},
-              {
-                autoEncryption: {
-                  keyVaultNamespace,
-                  kmsProviders: getKmsProviders(),
-                  schemaMap
-                }
-              }
-            );
-            return this.clientEncrypted.connect();
-          })
-      );
-    });
+            decrypted = await clientEncryption.decrypt(encrypted);
+          });
 
-    afterEach(function () {
-      if (this.commandStartedEvents) {
-        this.commandStartedEvents.teardown();
-        this.commandStartedEvents = undefined;
-      }
-      return Promise.resolve()
-        .then(() => this.clientEncrypted && this.clientEncrypted.close())
-        .then(() => this.client && this.client.close());
-    });
+          it('must create a data key', metadata, function () {
+            expect(dataKey).to.have.property('sub_type', 4);
+          });
 
-    it('should work for local KMS provider', metadata, function () {
-      let localDatakeyId;
-      let localEncrypted;
-      return Promise.resolve()
-        .then(() => {
-          // #. Call ``client_encryption.createDataKey()`` with the ``local`` KMS provider and keyAltNames set to ``["local_altname"]``.
-          // - Expect a BSON binary with subtype 4 to be returned, referred to as ``local_datakey_id``.
-          // - Use ``client`` to run a ``find`` on ``keyvault.datakeys`` by querying with the ``_id`` set to the ``local_datakey_id``.
-          // - Expect that exactly one document is returned with the "masterKey.provider" equal to "local".
-          // - Check that ``client`` captured a command_started event for the ``insert`` command containing a majority writeConcern.
-          this.commandStartedEvents.clear();
-          return this.clientEncryption
-            .createDataKey('local', { keyAltNames: ['local_altname'] })
-            .then(result => {
-              localDatakeyId = result;
-              expect(localDatakeyId).to.have.property('sub_type', 4);
-            })
-            .then(() => {
-              return this.client
-                .db(keyVaultDbName)
-                .collection(keyVaultCollName)
-                .find({ _id: localDatakeyId })
-                .toArray();
-            })
-            .then(results => {
-              expect(results)
-                .to.have.a.lengthOf(1)
-                .and.to.have.nested.property('0.masterKey.provider', 'local');
-              expect(this.commandStartedEvents.events).to.containSubset([
-                { commandName: 'insert', command: { writeConcern: { w: 'majority' } } }
-              ]);
+          it('properly encrypts and decrypts', metadata, function () {
+            expect(decrypted).to.equal('test');
+          });
+
+          it('fails with invalid provider host', metadata, async function () {
+            try {
+              await clientEncryptionInvalid.createDataKey('kmip', { masterKey });
```

Comment:
If this doesn't throw this test still passes, we should add an `expect.fail()` below this call

Reply:
Added.